server/socket/socket.js

```diff
@@ -1,14 +1,46 @@
+const cookieParser = require('socket.io-cookie-parser');
+const { authorization } = require('./authorization');
+
+const { Interview } = require('../models/Interview');
+
 const socketToUsers = new Map();
 const usersToSockets = new Map();
 const rooms = {};
+
 module.exports = (server) => {
-  const io = require('socket.io')(server, { origins: '*:*', path: '/sockets' });
+  const io = require('socket.io')(server, { path: '/sockets' });
+  io.use(cookieParser());
+  io.use(authorization);
+
   io.on('connection', (socket) => {
     const { id } = socket;
-    console.log('LOGGED IN! SOCKET ID', socket.id);
+
+    console.log(
+      `Connected and authorized, SOCKET ID: ${id} USERID: ${socket.userId}`
+    );
+    socketToUsers[id] = socket.userId;
+    usersToSockets[socket.userId] = socket.id;
+
+    socket.on('leave_interview', ({ roomId }) => {
+      console.log('second person leaving!!!!', Object.keys(rooms[roomId]));
+
+      if (Object.keys(rooms[roomId]).length === 2) {
```

Comment:
I'm a bit confused on the logic here. This always logs `second person leaving` but then conditionally logs `first person leaving`? What is `Object.keys(rooms[roomId]).length === 2` checking?

Reply:
the 3rd key is the code.  So if there are only 1 keys, there is only 1 person in the room.  If there are 3 keys, its full, if there is 1 key, there's nobody